Wrappers/Python/cil/optimisation/functions/TotalGeneralisedVariation.py

```diff
@@ -0,0 +1,211 @@
+        algorithm to solve:
+
+
+        .. math:: \mathrm{prox}_{\tau \mathrm{TGV}_{\alpha,\beta}}(b) := \underset{u}{\mathrm{argmin}} \frac{1}{2\tau}\|u - b\|^{2} + \mathrm{TGV_{\alpha, \beta}}(u) \Leftrightarrow
+
+        .. math:: \underset{u,w}{\mathrm{argmin}} \frac{1}{2\tau}\|u - b\|^{2} +  \alpha \|\nabla u - w\|_{2,1} + \beta\|\mathcal{E}w\|_{2,1} 
+
+        
+        The algorithm used for the proximal operator of TGV is the Primal Dual Hybrid Algorithm, see :class:`.PDHG`.
+
+
+        Parameters
+        ----------
+        max_iteration : :obj:`int`, default = 100
+            Maximum number of iterations for the PDHG algorithm.
+        correlation : :obj:`str`, default = `Space`
+            Correlation between `Space` and/or `SpaceChannels` for the :class:`.GradientOperator`.
+        backend :  :obj:`str`, default = `c`      
+            Backend to compute the :class:`.GradientOperator`  
+        split : :obj:`boolean`, default = False
+            Splits the Gradient into spatial gradient and spectral or temporal gradient for multichannel data.
+        verbose : :obj:`int`, default = 0
+            Prints primal, dual and primal-dual objectives of the :class:`PDHG` algorithm.
+        update_objective_interval : :obj:`int`, default = :code:`max_iteration`
+            Updates the objectives of the :class:`PDHG` algorithm.
+
+
+        Examples
+        --------
+
+        .. math:: \mathrm{prox}_{\tau\mathrm{TGV}_{\alpha, \beta}}(b) = \underset{u}{\mathrm{argmin}} \frac{1}{2\tau}\|u - b\|^{2} + \mathrm{TGV}_{\alpha, \beta}(u) 
+
+
+        >>> alpha1 = 1.0
+        >>> alpha0 = 2.0
+        >>> TGV = TotalGeneralisedVariation(alpha1 = alpha1, alpha0 = alpha0)
+        >>> sol = TGV.proximal(b, tau = 1.0)
+
+
+        """   
+        
+            
+    def __init__(self,
+                 reg_parameter=1.0,
+                 alpha1 = 1.0,
+                 alpha0 = 2.0,
+                 max_iteration = 100, 
+                 update_objective_interval = 50,
+                 correlation = "Space",
+                 backend = "c",
+                 split = False,
+                 verbose = 0,                  
+                 warmstart=False):
+        
+        super(TotalGeneralisedVariation, self).__init__(L = None)
+
+        # regularisation parameters for TGV
+        self.alpha1 = alpha1
+        self.alpha0 = alpha0
+        self.reg_parameter = reg_parameter
+                
+        # Iterations for PDHG_TGV
+        self.iterations = max_iteration
+        
+        # correlation space or spacechannels
+        self.correlation = correlation
+
+        # backend for the gradient
+        self.backend = backend        
+        
+        # splitting Gradient
+        self.split = split
+                        
+        self.verbose = verbose
+        self.update_objective_interval = update_objective_interval
+
+        # warm-start
+        self.warmstart  = warmstart
```

Comment:
Shouldn't it be `warm_start`? They are 2 words. https://tomographicimaging.github.io/CIL/nightly/developer_guide.html#other-methods
> All methods should follow the convention of small caps underscore separated words.

I notice that the same `warmstart` is used in  https://github.com/TomographicImaging/CIL/pull/1201